sdk/eventhub/azure-eventhub/azure/eventhub/aio/_error_async.py

```diff
@@ -14,23 +15,31 @@
     EventDataError
 )
 
+if TYPE_CHECKING:
+    from ._client_base_async import ClientBaseAsync, ConsumerProducerMixin
+
 _LOGGER = logging.getLogger(__name__)
 
 
-async def _handle_exception(exception, closable):  # pylint:disable=too-many-branches, too-many-statements
+async def _handle_exception(  # pylint:disable=too-many-branches, too-many-statements
+        exception: Exception,
+        closable: Union['ClientBaseAsync', 'ConsumerProducerMixin']
+    ) -> Exception:
+    # pylint: disable=protected-access
     if isinstance(exception, asyncio.CancelledError):
         raise exception
+    error = exception
     try:
-        name = closable._name  # pylint: disable=protected-access
+        name = cast('ConsumerProducerMixin', closable)._name  # type: ignore
```

Comment:
The point of the cast is to avoid a "type: ignore", so I'm surprised here. What do you get without the "type: ignore" here?

Reply:
ConsumerProducerMixin didn't have attribute _name. Its subclass had. So at runtime this worked. But not good readibility nor mypy friendly.
I added _name and a few other attributes to ConsumerProducerMixin so we don't need "type: ignore“ for these attributes referenced from ConsumerProducerMixin.